app/main.py

```diff
@@ -74,23 +37,23 @@ def create_centers_of_walls(field_size: tuple, wall_size: tuple):
 
 while running:
     for event in pygame.event.get():
-        if event.type == KEYDOWN:
-            if event.key == K_ESCAPE:
-                running = False
-
-        elif event.type == QUIT:
+        if (
+                event.type == KEYDOWN
+                and event.key == K_ESCAPE
+                or event.type != KEYDOWN
+                and event.type == QUIT
+        ):
             running = False
 
-    player.update()
+    player.update(walls, bombs, all_sprites)
```

Comment:
It's not a good idea to pass all sprites as arguments to the `update` method. It's better to create a class that stores all groups in it. You can pass it to the constructor of all sprites.

Reply:
I didn't get that yesterday.